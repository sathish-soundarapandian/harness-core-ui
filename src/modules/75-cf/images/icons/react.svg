<svg fill="none" height="32" viewBox="0 0 33 32" width="33" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><clipPath id="a"><path d="m.5 0h32v32h-32z"/></clipPath><g clip-path="url(#a)"><path d="m26.8104 10.977c-.3319-.1147-.6759-.2232-1.0301-.3258.0582-.2385.1116-.4739.1592-.70524.7797-3.79915.2699-6.85978-1.4711-7.86746-1.6694-.96624-4.3995.04122-7.1568 2.4497-.2651.23159-.531.4768-.7968.73378-.1771-.17003-.3539-.33433-.5302-.4915-2.8898-2.5753-5.7863-3.66057-7.52553-2.64996-1.66777.96904-2.16167 3.84631-1.45976 7.4468.06779.34777.14702.70288.23661 1.06398-.40993.1168-.80569.2413-1.18436.3738-3.38789 1.1855-5.55156 3.0436-5.55156 4.9709 0 1.9905 2.3227 3.987 5.85146 5.1976.27847.0955.56743.1858.86521.2716-.09666.3905-.18073.7732-.2513 1.1465-.66928 3.538-.14662 6.3473 1.51672 7.3102 1.71811.9944 4.60151-.0277 7.40921-2.4911.2219-.1947.4446-.4012.6677-.6177.2892.2795.5779.5441.8651.7922 2.7196 2.3489 5.4056 3.2975 7.0674 2.3319 1.7164-.9973 2.2742-4.0152 1.55-7.6869-.0553-.2804-.1196-.5669-.1919-.8585.2025-.0601.4013-.1221.5951-.1866 3.6685-1.2199 6.0553-3.1922 6.0553-5.2092 0-1.9343-2.2334-3.8049-5.6896-4.999zm-.7956 8.9058c-.175.0582-.3546.1144-.5375.1688-.405-1.2868-.9516-2.6552-1.6205-4.069.6383-1.38 1.1638-2.7309 1.5574-4.0094.3274.095.6451.1953.9513.3011 2.9612 1.0232 4.7675 2.536 4.7675 3.7017 0 1.2416-1.9508 2.8534-5.1182 3.9068zm-1.3143 2.614c.3202 1.6236.3659 3.0915.1538 4.239-.1906 1.0311-.5739 1.7186-1.0478 1.9939-1.0084.5859-3.165-.1757-5.4909-2.1845-.2666-.2303-.5351-.4762-.8046-.7364.9017-.9898 1.8029-2.1405 2.6824-3.4184 1.5469-.1378 3.0085-.3631 4.3339-.6706.0652.2643.1232.5235.1732.777zm-13.291 6.1317c-.9853.3492-1.77.3592-2.24435.0847-1.00938-.5843-1.42901-2.8399-.85662-5.8656.06556-.3465.1436-.7025.23353-1.0663 1.31087.2909 2.76174.5003 4.31234.6265.8855 1.2504 1.8126 2.3999 2.747 3.4063-.2041.1978-.4075.3861-.6097.5636-1.2415 1.0892-2.4857 1.8621-3.5822 2.2508zm-4.61592-8.7533c-1.56035-.5353-2.84895-1.231-3.73222-1.9901-.79368-.6823-1.19439-1.3596-1.19439-1.9091 0-1.1696 1.73721-2.6613 4.63457-3.6752.35156-.1231.71958-.239 1.10149-.3479.40042 1.3073.92571 2.6742 1.55961 4.0564-.64211 1.4028-1.17485 2.7915-1.5792 4.1141-.27227-.0785-.53605-.1612-.78986-.2482zm1.5473-10.57138c-.60138-3.08481-.20198-5.41187.80308-5.99581 1.07054-.62208 3.43784.26487 5.93294 2.48851.1595.14213.3196.29089.4801.44479-.9297 1.00203-1.8484 2.14281-2.726 3.38592-1.505.14001-2.9456.36492-4.27014.66607-.08327-.33619-.15704-.66653-.21998-.98948zm13.80332 3.42098c-.3166-.5489-.6417-1.0849-.9731-1.6062 1.021.1295 1.9993.3015 2.9179.5119-.2758.8872-.6195 1.8148-1.0242 2.7657-.2902-.5531-.5973-1.1108-.9206-1.6714zm-5.6289-5.5029c.6305.68564 1.262 1.45116 1.883 2.28173-.6258-.02967-1.2594-.04511-1.8983-.04511-.6328 0-1.2618.0151-1.8838.04425.6217-.82285 1.2587-1.58734 1.8991-2.28087zm-5.6651 5.5124c-.3163.5505-.6176 1.1052-.90328 1.6609-.39809-.9476-.73866-1.8794-1.01685-2.7792.91301-.205 1.88663-.3727 2.90113-.4998-.3361.5261-.6638 1.066-.981 1.618zm1.0102 8.1993c-1.0482-.1174-2.0364-.2764-2.94918-.4758.28263-.9158.63071-1.8675 1.03723-2.8356.28645.5553.58895 1.1102.90715 1.6619h.0001c.3242.5621.66 1.1126 1.0047 1.6495zm4.6927 3.8932c-.6479-.7017-1.2942-1.4778-1.9254-2.3127.6128.0241 1.2374.0365 1.8723.0365.6522 0 1.297-.0148 1.9317-.0431-.6232.8499-1.2523 1.6272-1.8786 2.3193zm6.5248-7.2539c.4277.9785.7882 1.9252 1.0745 2.8254-.9278.2125-1.9296.3836-2.985.5107.3322-.5283.6601-1.0735.9823-1.6343.326-.5674.6355-1.1356.9282-1.7018zm-2.1123 1.0163c-.5002.8706-1.0138 1.7016-1.5351 2.486-.9496.0682-1.9306.1033-2.9305.1033-.9958 0-1.9645-.0311-2.8972-.0918-.5424-.7948-1.0671-1.6283-1.5642-2.49h.0002c-.4958-.8594-.952-1.726-1.3654-2.5878.4133-.8638.8683-1.7314 1.3612-2.5893l-.0001.0002c.4942-.8601 1.0143-1.6896 1.5516-2.4793.9517-.0722 1.9276-.1099 2.9138-.1099h.0001c.9907 0 1.9678.038 2.9192.1108.5292.784 1.0458 1.6107 1.5419 2.471.5017.8699.9626 1.7317 1.3795 2.577-.4156.8598-.8755 1.7303-1.375 2.5998zm2.82-15.32201c1.0716.62024 1.4883 3.12168.815 6.40183-.0429.2093-.0913.42248-.1438.63858-1.3276-.3075-2.7693-.5363-4.2786-.67842-.8793-1.25678-1.7905-2.39934-2.7051-3.3885.2459-.23743.4914-.46382.736-.67747 2.3624-2.06357 4.5705-2.87831 5.5765-2.29602zm-7.2856 9.84071c1.5781 0 2.8575 1.284 2.8575 2.8681 0 1.5839-1.2794 2.868-2.8575 2.868s-2.8575-1.2841-2.8575-2.868c0-1.5841 1.2794-2.8681 2.8575-2.8681z" fill="#00d8ff"/></g></svg>