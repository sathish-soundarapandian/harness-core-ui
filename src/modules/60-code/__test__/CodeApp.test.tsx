/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */
import React from 'react'
import { render } from '@testing-library/react'
import {
  Repository,
  Repositories,
  Commits,
  Branches,
  FileEdit,
  Settings,
  PullRequests,
  PullRequest,
  Compare,
  Webhooks,
  WebhookNew,
  WebhookDetails,
  Tags
} from '../CodeApp'

// eslint-disable-next-line jest-no-mock
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: jest.fn()
  }),
  useRouteMatch: () => ({ params: { accountId: '1234', orgIdentifier: 'abc', projectIdentifier: 'xyz' } })
}))

jest.mock(
  'react',
  () => ({
    ...jest.requireActual('react'),
    Suspense: ({ children }: { children: JSX.Element }) => <div>{children}</div>
  }),
  { virtual: true }
)

jest.mock('framework/strings', () => ({
  useStrings: () => ({
    getString: (id: string) => id
  })
}))

jest.mock(
  'code/App',
  () => ({
    default: ({ children }: { children: JSX.Element }) => <div>code/App {children}</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Repositories',
  () => ({
    default: () => <div>code/Repositories</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Repository',
  () => ({
    default: () => <div>code/Repository</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/FileEdit',
  () => ({
    default: () => <div>code/FileEdit</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Commits',
  () => ({
    default: () => <div>code/Commits</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Branches',
  () => ({
    default: () => <div>code/Branches</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Tags',
  () => ({
    default: () => <div>code/Tags</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/PullRequests',
  () => ({
    default: () => <div>code/PullRequests</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Compare',
  () => ({
    default: () => <div>code/Compare</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/PullRequest',
  () => ({
    default: () => <div>code/PullRequest</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Settings',
  () => ({
    default: () => <div>code/Settings</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/Webhooks',
  () => ({
    default: () => <div>code/Webhooks</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/WebhookNew',
  () => ({
    default: () => <div>code/WebhookNew</div>
  }),
  { virtual: true }
)

jest.mock(
  'code/WebhookDetails',
  () => ({
    default: () => <div>code/WebhookDetails</div>
  }),
  { virtual: true }
)

describe('CodeApp', () => {
  test('Exports should work properly', async () => {
    const { container } = render(
      <div>
        <Repository />
        <Repositories />
        <Commits />
        <Branches />
        <Tags />
        <FileEdit />
        <Settings />
        <PullRequests />
        <PullRequest />
        <Compare />
        <Webhooks />
        <WebhookNew />
        <WebhookDetails />
      </div>
    )

    expect(container).toMatchSnapshot()
  })
})
