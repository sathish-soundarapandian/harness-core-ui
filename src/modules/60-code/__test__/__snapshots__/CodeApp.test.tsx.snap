// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeApp Exports should work properly 1`] = `
<div>
  <div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
    <div>
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main StyledProps--padding-large"
      >
        <p
          class="StyledProps--font StyledProps--main"
        >
          Something went wrong...
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          This error has been reported and we are looking into it with high priority.
        </p>
        <p
          class="StyledProps--font StyledProps--main"
        >
          Please 
          <a
            href="#"
          >
            refresh
          </a>
           your browser to continue.
        </p>
      </div>
    </div>
  </div>
</div>
`;
